src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2094,7 +2100,14 @@ private unsafe ParsingError PrivateParseMinimal()
                                 }
                             }
                         }
-                        else if (_syntax.InFact(UriSyntaxFlags.FileLikeUri) && (i - idx >= 2 && i - idx != 3 &&
+                        //Unix: Unix path?
+                        else if (!IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && (i - idx >= 2))
+                        {
+                            idx += 2;
+                            _flags |= Flags.UnixPath;
+                        }
+                        //Windows: UNC path?
+                        else if (IsWindowsSystem && _syntax.InFact(UriSyntaxFlags.FileLikeUri) && (i - idx >= 2 && i - idx != 3 &&
```

Comment:
I believe Roslyn will effectively trim away the dead branches based on it seeing that IsWindowsSystem is hardcoded to be true or false.  Can you double-check just to confirm?  If it doesn't, we may want to rethink this.